File: 0343-integer-break/0343-integer-break.js
/**
 * @param {number} n
 * @return {number}
 */
var integerBreak = function(n) {
  if (n < 4) return n - 1;
  
  const [avg, remain] = [~~(n / 3), n % 3];
  if (!remain) return 3 ** avg;
  if (remain === 1) return 3 ** (avg - 1) * 4;
  
  return 3 ** avg * 2;
};

File: 0343-integer-break/0343-integer-break.py
class Solution:
    def integerBreak(self, n: int) -> int:
        if n < 4:
            return n - 1
          
        avg, remain = n // 3, n % 3
        if remain == 0:
            return 3 ** avg
        if remain == 1:
            return 3 ** (avg-1) * 4
        return 3 ** avg * 2